onnxruntime/python/tools/transformers/convert_generation.py

```diff
@@ -867,6 +876,211 @@ def move_initializers(
     return moved_initializers
 
 
+def update_input_shapes_for_gpt2_decoder_model(decoder_onnx_path: str, use_external_data_format: bool = True):
+    """Update the input shapes for the inputs "input_ids" and "position_ids" and make the sequence length dim value 1 for each of them.
+       The decoder model will be over-written.
+
+    Args:
+        decoder_onnx_path (str): Path of GPT-2 decoder onnx model
+        use_external_data_format(bool): output tensors to external data or not.
+    """
+
+    decoder_model_proto = onnx.load_model(decoder_onnx_path, load_external_data=True)
+    for i in range(len(decoder_model_proto.graph.input)):
+        if (
+            decoder_model_proto.graph.input[i].name == "input_ids"
+            or decoder_model_proto.graph.input[i].name == "position_ids"
+        ):
+            decoder_model_proto.graph.input[i].type.tensor_type.shape.dim[1].dim_value = 1
```

Comment:
clear dim_param if exists. When dim_param and dim_value both exist, dim_param will be used.